File: backend/Models/Order.cs
namespace backend.Models;

public class Order
{
    public int OrderId { get; set; }
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    public List<OrderItem> OrderItems { get; set; } = new();
    public string Status { get; set; } = "pending";
    public decimal TotalAmount { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/order.ts
export type OrderItem = {
  productId: number;
  productName: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
};

export type Order = {
  orderId: number;
  status: string;
  totalAmount: number;
  items: OrderItem[];
  createdAt: string;
};
